twitterMini/Twitter.swift

```diff
@@ -0,0 +1,130 @@
+    
+    func authorizeTwitter(completion: (success: Bool) -> Void){
+        if let callbackURL = NSURL(string: callback) {
+            oauthManager.authorizeWithCallbackURL( callbackURL, success: {
+                credential, response, parameters in
+                print("Authentication Success, userID:\(parameters["user_id"])")
+                self.idForLoggedInUser = parameters["user_id"]
+                completion(success:true)
+                }, failure: { error in
+                    print("Authentication Error: \(error.localizedDescription)")
+                    completion(success:false)
+                }
+            )
+        }
+    }
+
+    func fetchTimeline(completion: (rawStatuses: [[String:AnyObject]]?, error: NSError?) -> Void){
+        if let idForLoggedInUser = idForLoggedInUser{
+            let homeTimelineEndpoint = "statuses/home_timeline.json"
+            let homeTimelineURL = "\(twitterBaseURL)\(homeTimelineEndpoint)?user_id=\(idForLoggedInUser)"
+            print("hitting \(homeTimelineURL)")
+            oauthManager.client.get(homeTimelineURL, success: { data, response in
+                print("Fetch Home Timeline Success")
+                do {
+                    let dataObject = try NSJSONSerialization.JSONObjectWithData(data, options: []) as? [[String:AnyObject]]
+                    completion(rawStatuses: dataObject, error: nil)
+                } catch {
+                    print("Error Serializing JSON: \(error)")
+                    completion(rawStatuses: nil, error: error as NSError)
+                }
+                }, failure: { error in
+                    print("Fetch Home Timeline Error:\(error.localizedDescription)")
+                    completion(rawStatuses: nil, error: error)
+            })
+        } else {
+            print("user is not logged in...")
+            // FIXME: use constants
+            let error = NSError(domain: "Tom's Domain", code: 420, userInfo: [NSLocalizedDescriptionKey:"There is no authenticated user."])
+            completion(rawStatuses:nil, error: error)
+        }
+    }
+}
+
+// MARK: Data Manager
+
+class TwitterDataManager {
+    static let sharedManager = TwitterDataManager()
+    let twitter = TwitterAPIClient.sharedClient // maybe make just the OAuth1Swift a singleton and use that to init any API instances?
+    var statuses = [TwitterStatus]()
+    
+    func authorize(completion:(success: Bool) -> Void){
+        twitter.authorizeTwitter { success in
+            completion(success: success)
+        }
+    }
+    
+    func populateStatuses(completion: (NSError?) -> Void){
+        twitter.fetchTimeline { rawStatuses, error in
+            if let error = error {
+                completion(error)
+            } else {
+                // FIXME: use genome
+                for statusDictionary in rawStatuses ?? [] {
+                    if let createdAt = statusDictionary["created_at"] as? String,
```

Comment:
You could try using https://github.com/LoganWright/Genome for your object serialization. It's pretty advanced but it eliminates of of this boilerplate code